lua/telescope/utils.lua

```diff
@@ -213,16 +213,30 @@ utils.path_smart = (function()
   end
 end)()
 
+-- vim.fn.fnamemodify(path, ":p:t") may replace util.path_tail(path),
+-- but the former may be slower and has dependency on neovim
```

Comment:
nit: this "dependency on neovim" comment isn't really necessary.
This whole plugin is basically entirely dependent on neovim.

also out of curiosity, I benchmarked `fnamemodify` vs `path_tail` against ~80k files.
```
hyperfine 'nvim -l fnamemodify.lua' 'nvim -l path_tail.lua' --warmup 10 --runs 10
Benchmark 1: nvim -l fnamemodify.lua
  Time (mean ± σ):     238.0 ms ±   4.3 ms    [User: 119.9 ms, System: 108.0 ms]
  Range (min … max):   232.3 ms … 243.1 ms    10 runs

Benchmark 2: nvim -l path_tail.lua
  Time (mean ± σ):      28.8 ms ±   3.0 ms    [User: 21.9 ms, System: 6.3 ms]
  Range (min … max):    25.5 ms …  34.8 ms    10 runs

Summary
  nvim -l path_tail.lua ran
    8.27 ± 0.88 times faster than nvim -l fnamemodify.lua
```
kinda surprised by this result.

Reply:
I thought it would be only 2 to 3 times faster. `vim.fn` calls `viml` calls c code, probably that's why it's much slower than pure lua codes.